maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java

```diff
@@ -304,23 +323,22 @@ public String toString() {
      * Represents a string in the version item list, usually a qualifier.
      */
     private static class StringItem implements Item {
-        private static final List<String> QUALIFIERS =
-                Arrays.asList("alpha", "beta", "milestone", "rc", "snapshot", "", "sp");
+        private static final List<String> QUALIFIERS = Arrays.asList("snapshot", "", "sp");
```

Comment:
So this change relies on the lexicographical order or the tokens, right?

Reply:
yes, it relies on the lexicographical order